src/hotspot/cpu/riscv/assembler_riscv.hpp

```diff
@@ -411,8 +410,55 @@ class Assembler : public AbstractAssembler {
     return -1;
   }
 
+  static int zfa_zli_lookup_half_float(uint16_t value) {
+    switch(value) {
+      case 0xbc00 : return  0;
+      case 0x0400 : return  1;
+      case 0x0100 : return  2;
+      case 0x0200 : return  3;
+      case 0x1c00 : return  4;
+      case 0x2000 : return  5;
+      case 0x2c00 : return  6;
+      case 0x3000 : return  7;
+      case 0x3400 : return  8;
+      case 0x3500 : return  9;
+      case 0x3600 : return 10;
+      case 0x3700 : return 11;
+      case 0x3800 : return 12;
+      case 0x3900 : return 13;
+      case 0x3a00 : return 14;
+      case 0x3b00 : return 15;
+      case 0x3c00 : return 16;
+      case 0x3d00 : return 17;
+      case 0x3e00 : return 18;
+      case 0x3f00 : return 19;
+      case 0x4000 : return 20;
+      case 0x4100 : return 21;
+      case 0x4200 : return 22;
+      case 0x4400 : return 23;
+      case 0x4800 : return 24;
+      case 0x4c00 : return 25;
+      case 0x5800 : return 26;
+      case 0x5c00 : return 27;
+      case 0x7800 : return 28;
+      case 0x7c00 : return 29;
+      // case 0x7c00 : return 30; // redundant with 29
+      case 0x7e00 : return 31;
```

Comment:
This switch-case for all H/S/D variants seems nontrivial. One possible enhancement to speed up might be putting the values in a table and doing a binary search.

Reply:
In this case, I think the compiler (gcc or clang) will be able to automatically generate a binary research rather than a linear search, so we don't have to write the code manually. And keep the code as it is is more readable.

Reply:
Interesting! I tried several recent GCC versions and I see it is doing a binary search. Then I think we are OK with the current shape. Thanks.